### DISCUSSION TITLE
The dream is finally on the horizon — seamless workflow with Lightroom (in 1-2 years?)

### DISCUSSION THREAD

**Discussion by develar** [immich-app/immich#8967] [General]
This post serves two main purposes: 
1. List issues that are blocking the inclusion of Immich into my workflow.
2. Learn about alternative photo management approaches.

Even though my wife and I are not professional photographers but hobbyists, we still shoot in RAW and post-process some of our photos. Lightroom can be a great tool with its AI-based search (which makes it easy to find specific pictures amongst 20+ photos of the same subject) and outstanding RAW editor. However, a few things make it less than perfect:

### Shortcomings of Lightroom

* **Lack of auto-creation of memories**: Unlike Apple Photos, Lightroom doesn't automatically create "memories".
    > In the For You tab, Photos searches your photos and videos to find standout moments and presents them in [collections called Memories.](https://support.apple.com/HT207023)  

    Automatic creation of memories can inspire more travels (e.g., "why do all our photos from last year show only one island?") and help reflect on your life and experiences.

* **Cost of storage**: The price tag of 12 euros for 1TB could mount up, especially considering increasing photo and video resolutions.

While Immich may not be a complete solution, especially considering it will not be a powerful RAW editor, I believe it can be combined effectively with Lightroom. 
Despite its current limitations, I see the potential in Immich.

### Proposed Workflow (Lightroom + Immich)

Here's how I envision the workflow:

1. Store all your data on your storage device or any cloud provider, such as pCloud.
2. Use Lightroom 7+ (which can work without Adobe Cloud, per articles [here](https://www.lightroomqueen.com/should-i-move-from-lightroom-classic-to-lightroom-local/) and [here](https://www.lightroomqueen.com/community/threads/lightroom-and-local-storage-option.49675/)) to edit files on your local disk. Your edits (post-process) are stored as XMP sidecar files. There is no a "version" feature, but you can create multiple copies if needed.
3. Connect your storage to Immich as an [external library](https://immich.app/docs/guides/external-library/).

Eventually, you get the best of both worlds: an exceptional AI-based RAW editor from Lightroom and a feature-rich image browser from Immich. 

Currently, a few issues with Immich prevent this workflow from being fully functional. I am optimistic that these will be solved within 1-2 years, which is why I became a GitHub sponsor and recommend others to do the same.

### Known immich Issues

* The re-scan of an external library [does not clear the queue](https://github.com/immich-app/immich/issues/8760). In my case, with 44K photos, each force re-scan creates new tasks for thumbnails, smart search, etc. (I use "force re-scan" as immich do not correctly read a creation time for videos exported by Lightroom, so, I tried several solutions to overcome it).
* [Stack raw+jpeg](https://github.com/immich-app/immich/discussions/2479) is not currently supported. Ideally, Immich should compute photo identity via RAW (as it is an immutable original), and allow the JPEG to change as a result of post-processing. 
  It's not clear to me whether it is possible to render RAW in the same way as Lightroom does, using an XMP sidecar. If it is, then there is no need to produce a JPEG using Lightroom; using RAW files alone would suffice.

Memories — https://github.com/immich-app/immich/discussions/2836 and https://github.com/immich-app/immich/discussions/7437 I do not count it as an issue, as it doesn't block me, opposite to issues above.

**Comment by AngelaDMerkel** [comment]
I am very strongly of the opinion that these tools are for entirely different purposes. RAW Developers make images and Immich/gPhotos display images. Exporting raster images from Lightroom into a specific folder which Immich reads as an external library is a much less complex and much more seamless workflow. 

Every single RAW Developer including LibRAW interprets raw images and XMP data completely differently. Edits you make in Lightroom will not be reflected accurately in Immich or any other RAW Developer of any kind, expect the one which make them. If you want an image to look a certain way export it is a JPEG or similar and have Immich read from that library.

**Reply by mertalev** [reply]
This is true. Our RAW support is meant more for images you don't care to edit. If you have specific expectations of how it should look, exporting it is your best option. You can also embed the export into the RAW file if you'd like to back up both with a single file.

**Reply by develar** [reply]
Yes, you are right. As I said, "It's not clear to me whether it is possible to render RAW in the same way as Lightroom does, using an XMP sidecar.".

Immich does the job in any case. I can use two directories (one for originals and another for export). 

Why I don't want to use only exported images for Immich and why support for stack RAW+rendered is essential — my relative shares photos with another relative, and the ability to download RAW for editing is wanted. It is one of the points of why I am so excited about Immich — I don't need to wait for Apple/Adobe to do what I need; I can try to implement what I need myself (I can do simple pull requests, but I cannot develop my photo management solution).

> You can also embed the export into the RAW file if you'd like to back up both with a single file.

Yes, I can convert RAW to DNG; in this case, everything will be packed into one file (the issue is that Lightroom doesn't allow me to. control the quality and size of the embedded file). Anyway, I would prefer to avoid conversion.

**Reply by AngelaDMerkel** [reply]
@develar in the meantime if you have a good reverse proxy or port forward solution it might be easiest to just offer SFTP or other managed access to your RAW images. It's generally not the best practice to have your originals managed by two applications, and you might find this frustrating if in the future you start to take advantage of _enhance details_ which creates duplicate (linear DNG, AI enhanced) images alongside the originals. How should Immich consider those?

**Reply by develar** [reply]
> How should Immich take these into account?

In the same way, with Stack. The way Lightroom does it (if you create a panorama from multiple photos, there is an option to create a stack).
It is probably possible to do this automatically, based on the EXIF metadata.

Therefore, "Immich should compute photo identity by RAW (as it is an immutable original)" — an immutable original serves as the identity for the asset.

**Comment by mertalev** [comment]
> I use "force re-scan" as immich do not correctly read a creation time for videos exported by Lightroom, so, I tried several solutions to overcome it

Can you elaborate on this? This could be a relatively simple fix on our end.

**Reply by develar** [reply]
> This could be a relatively simple fix on our end.

I am not sure. :) I do not yet have time to prepare a good bug report about this. Yep, I managed to fix the issue without re-exporting (all videos in Immich in a timeline for the day of export).
 I fixed EXIF tags/file modification times to align with what Immich uses and expects, but still, I am not 100% sure about correctness. I have 3500 videos from various cameras in various formats for ten years, so the fix is not easy or safe.
Maybe in 1-3 weeks, I will report the bug or send a pull request.

Slightly offtopic —
Even Lightroom cannot show the date for videos from Sony IV with the correct timezone (thanks to Sony) — https://www.lightroomqueen.com/community/threads/importing-sony-video-files-have-wrong-date-time-stamps.44417/ or https://community.adobe.com/t5/lightroom-classic-bugs/p-video-capture-time-shifted-by-time-zone-offset/idi-p/12249887

**Reply by mertalev** [reply]
Oh, it's a time zone issue... Maybe it was naive to think anything to do with exif and time would be simple haha

**Reply by AngelaDMerkel** [reply]
@mertalev Google Photos has this same issue on occasion. When there is no location metadata and no timezone offset (eg +02:00) Google assumes that the time is GMT and it causes no end of problems. Exiftool has a tool for this though, which sets the offset to whatever the users machines is currently in `-api QuicktimeUTC` 

This usually alleviates most time related issues, but especially on older devices, metadata tends to be garbled garbage. I have an older handycam whose video metadata is so rough I needed to write metadata in multiple operations to get it correct. 

```
#!/bin/bash

# Converts .mod MPEG-1 files to h265 mp4 and transforms to BT709 from BT470/BT490
# Should preserve date metadata
# Adds Camera exif

for file in *.MTS; do
    # Check for .MTS files
    [ -e "$file" ] || continue

    # Extract the filename without extension
    filename_no_ext="${file%.*}"

    # ffmpeg
    ffmpeg -i "$file" -map_metadata 0 -movflags use_metadata_tags -c:v copy -c:a copy "${filename_no_ext}.mp4"

    # Metadata transfer
    exiftool -overwrite_original -TagsFromFile "$file" -all:all "${filename_no_ext}.mp4"

    # Date and Time? 
    exiftool -overwrite_original "-TagsFromFile" "$file" "-DateTime<DateTimeOriginal" "-DateTimeDigitized<DateTimeOriginal" "-DateTimeOriginal<DateTimeOriginal" "${filename_no_ext}.mp4"

    #Camera Metadata
    exiftool -overwrite_original -m "-AllDates<DateTimeOriginal" -api QuicktimeUTC -Make="Sony" -Model="HDR-CX110" -LensMake="Carl Zeiss" -LensModel="Vario-Tessar 2.5-62.5mm f1.8" -ApertureValue="1.8" -FNumber="1.8" -Lens="Carl Zeiss Vario-Tessar 2.5-62.5mm f1.8" "${filename_no_ext}.mp4"
done
```
